com.unity.cinemachine/Editor/Editors/CinemachineCameraEditor.cs

```diff
@@ -29,22 +33,35 @@ static void AdoptSceneViewCameraSettings(MenuCommand command)
             cam.Lens = CinemachineMenu.MatchSceneViewCamera(cam.transform);
         }
 
-        void OnEnable()
+        /// <summary>
+        /// Called when the inspector is initially displayed.  Subclasses should make sure 
+        /// to call the base class implementation.
+        /// </summary>
+        protected virtual void OnEnable()
         {
             Undo.undoRedoPerformed += ResetTarget;
 
             CinemachineSceneToolUtility.RegisterTool(typeof(FoVTool));
             CinemachineSceneToolUtility.RegisterTool(typeof(FarNearClipTool));
         }
 
-        void OnDisable()
+        /// <summary>
+        /// Called when the inspector is dismissed.  Subclasses should make sure 
+        /// to call the base class implementation.
+        /// </summary>
+        protected virtual void OnDisable()
         {
             Undo.undoRedoPerformed -= ResetTarget;
```

Comment:
Then they get an ugly default inspector

Reply:
And a memory leak?

Reply:
The IDE will warn you if your private OnDisable is hiding the virtual one

Reply:
Subclassing CinemachineCamera is expected to be a very rare thing.  With this PR, a decent inspector is possible, otherwise it's not.  There's only so much handholding I'm willing to do.

Reply:
Let's make another ticket for the task of improving the experience of making editors for subclasses of CM stuff.  There are some ways we can explore to make it easier.

Reply:
This is probably a common issue. There may be a pattern already within unity to make this more robust. How can we find out?

I am ok with it as is for now.

Reply:
> Let's make another ticket for the task of improving the experience of making editors for subclasses of CM stuff. There are some ways we can explore to make it easier.

It's better than what we had, I agree we could merge has is.